omnipaxos/src/util.rs

```diff
@@ -337,7 +337,7 @@ impl SequenceNumber {
     pub(crate) fn check_msg_status(&self, msg_seq_num: SequenceNumber) -> MessageStatus {
         if msg_seq_num == SequenceNumber::PREDEFINED_LEADER_FIRST_ACCEPT {
             MessageStatus::First
-        } else if msg_seq_num.session < self.session {
+        } else if msg_seq_num <= *self {
```

Comment:
I guess this is correct, but it looks a bit strange that we ignore a message even if it is in the right session. I think this is fine if we move the `else if` of the `expected` case above this just for clearness.

Reply:
This was done to protect against duplicated messages. So even in the same session there are messages to ignore. One example is when we resend `AcceptStopSign` it has the same sequence number but we don't want to trigger a `PrepareReq`.

Reply:
I understand that, it's just a bit strange to read the code and see this if clause before the next one (the expected case) which is `else if msg_seq_num.session == self.session && ...`. That's why I think you should put the `else if` for the `Expected` case above this one.